ocaml/xapi/xapi_vdi.ml

```diff
@@ -878,11 +878,17 @@ let get_nbd_info ~__context ~self =
     raise (Api_errors.Server_error (Api_errors.vdi_incompatible_type, [ Ref.string_of self; Record_util.vdi_type_to_string `cbt_metadata ]))
   end;
 
+  (* Ignore exceptions that can happen due to invalid references and just skip them *)
+  let default_on_exn f default x = try f x with _ -> default in
```

Comment:
Good question. The reason is that I thought this call just returns exceptions related to invalid refs so catching everything isn't harmful, plus I did not know what exceptions are thrown for invalid refs. But I agree, if it can throw other exceptions as well, than we should just catch the ones related to invalid refs.